packages/labs/eleventy-plugin-lit/src/index.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const path = require('path');
+
+type LitPluginOptions = {
+  componentModules?: string[];
+};
+
+module.exports = {
+  configFunction: function (
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    eleventyConfig: any,
+    options: LitPluginOptions = {}
+  ) {
+    eleventyConfig.addTransform(
+      'render-lit',
+      async (content: string, outputPath: string) => {
+        console.log(this);
+        if (outputPath.endsWith('.html')) {
+          const render = (
+            await import('@lit-labs/ssr/lib/render-with-global-dom-shim.js')
+          ).render;
+          const html = (await import('lit')).html;
+          const unsafeHTML = (await import('lit/directives/unsafe-html.js'))
+            .unsafeHTML;
+          for (const module of options.componentModules ?? []) {
+            require(path.join(process.cwd(), module));
```

Comment:
This could be a dynamic `import()`, or maybe `componentModules` could be an array of `{path: string, type: 'module' | 'commonjs'}` to be able to choose the import type. That might not even be necessary because `import()` works for both modules and CommonJS, but I'm not sure what happens with the require cache if you `import()` CommonJS.

Reply:
I'm going to be switching to vm modules in a followup.